src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketReviewer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016-2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket.api;
+
+/**
+ * Represents the reviewer of a pull request.
+ */
+public interface BitbucketReviewer {
```

Comment:
Is this really needed? Something in the json parser that requires it? Seems to me like a standard pojo would be enough otherwise.

Reply:
There are two implementations of this interface. `BitbucketReviewerValue` for Bitbucket Cloud and `BitbucketServerReviewer` for Bitbucket Server. The way that `getReviewerLogin` is implemented is different in cloud and server unfortunately because the field names are different in the JSON (`user.name` vs `user.username`).

Reply:
Not needed, we can update to Jackson 2.9 and use [`JsonAlias`](https://www.concretepage.com/jackson-api/jackson-jsonproperty-and-jsonalias-example#JsonAlias)